EIPS/eip-7543.md

```diff
@@ -0,0 +1,158 @@
+---
+eip: 7543
+title: EVM decimal math
+description: This EIP adds OPCODEs to allow high precision decimal float calculation of all elementary functions with precise gas enumeration.
+author: 1m1 (@1m1-github)
+discussions-to: https://ethereum-magicians.org/t/decimal-math-on-evm/16194
+status: Draft
+type: Standards Track
+category: Core
+created: 2023-10-22
+---
+
+
+## Abstract
+
+This EIP adds *decimal float* OPCODEs for arithmetic via DECADD, DECNEG, DECMUL, DECINV and expression of all elementary functions via DECEXP, DECLN, DECSIN. All decimal values upto the maximal precision allowed by a int256 coefficient and exponent are represented exactly, as c*10^q. All implemented algorithms converge for all inputs given enough precision, as chosen by the user. All calculations are deterministic and gas is precisely embedded bottom-up. Allowing high precision decimal elementary functions invites the worlds of mathematical finance, machine learning, science, digital art, games and others to Ethereum. The implementation is functional.
+
+## Motivation
+
+Currently, to take a power, a^b, of non integer values, requires vast amounts of Solidity code.
+The simplest task in trading e.g. is to convert volatilities from yearly to daily, which involves taking the 16th root.
+
+Giving users/devs the same ability that scientific calculators have allows for the creation of apps with higher complexity.
+
+The files [BlackScholes.yul](../assets/eip-7543/BlackScholes.yul) and [Neuron.yul](../assets/eip-7543/Neuron.yul) demonstrate how these OPCODEs simplify numerical smart contract code.
+
+### Why decimal?
+
+A simple value like 0.1 cannot be represented finitely in binary. Decimal types are much closer to the vast majority of numerical calculations run by humans.
+
+### eVm
```

Comment:
rename this to Rational

Reply:
ahh i mean rename the section to `Rationale` rather than `Rational numbers` my bad for the typo

Reply:
which section exactly? i see this is within a `Motivation` section and further below there is a `Rationale` section.